Sources/SpeziViews/ViewModifier/ShimmerModifier.swift

```diff
@@ -0,0 +1,68 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2024 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import SwiftUI
+
+
+/// A view modifier that applies a shimmer animation effect.
+///
+/// The `ShimmerViewModifier` applies a simple opacity animation that creates a shimmer effect.
+/// It can be useful when displaying placeholders for content that is being asynchronously loaded.
+/// The `SkeletonLoadingModifier` combines this with a vertical replication of the view to create a skeleton loading effect.
+///
+/// ### Usage
+///
+/// ```swift
+/// struct ShimmerModifierTestView: View {
+///     @State var loading = false
+///     let secondaryColor = Color(.init(gray: 0.8, alpha: 1.0))
+///
+///     var body: some View {
+///         VStack {
+///             ExampleAsyncView(loading: $loading)
+///                 .processingOverlay(isProcessing: loading) {
+///                     RoundedRectangle(cornerRadius: 10)
+///                         .fill(secondaryColor)
```

Comment:
We could use something like `.gray.opacity(0.3)` instead here? Might be better than the constant above.